src/main/java/harmony/dev/harmonyserver/controller/MemberController.java

```diff
@@ -0,0 +1,38 @@
+package harmony.dev.harmonyserver.controller;
+
+import harmony.dev.harmonyserver.DTO.MemberFindDTO;
+import harmony.dev.harmonyserver.domain.Member;
+import harmony.dev.harmonyserver.service.MemberService;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.Optional;
+
+@RestController
+public class MemberController {
+
+    private final MemberService memberService;
+
+    public MemberController(MemberService memberService) {
+        this.memberService = memberService;
+    }
+
+    @PostMapping("/member/join")
+    @ResponseBody
+    public String join(@Valid @RequestBody Member member){
+        return memberService.join(member);
+    }
+
+    @PostMapping("/member/find")
+    @ResponseBody
+    public Optional<Member> find(@Valid @RequestBody MemberFindDTO memberFindDTO){
+        if(memberFindDTO.getType().equals("user_id")){
+            return memberService.findByUserId(memberFindDTO.getKey());
+        }else if(memberFindDTO.getType().equals("phone_number")){
+            return memberService.findByPhoneNumber(memberFindDTO.getKey());
+        }else{
+            throw new IllegalStateException("잘못된 타입입니다.");
```

Comment:
```suggestion
        String dtoType = memberFindDTO.getType();
        Optional<Member> result = switch(dtoType) {
            case "user_id" -> memberService.findByUserId(memberFindDTO.getKey());
            case "phone_number" -> memberService.findByPhoneNumber(memberFindDTO.getKey());
            default -> null;
        }
        if(result == null) {
            throw new IllegalStateException("잘못된 타입입니다.");
        }
        return result;
```
위와 같은 형태로 변경하는 것이 어떤지
- getType() 1회만 수행
- 마지막에 return. 필요한 경우 return 이전에 find 결과에 대한 추가 작업 수행 가능

또는 MemberRepository의 find가 하나로 통합된다면 join과 마찬가지로
`return memberService.find(...);` 형태로 변경하고,
위 로직은 함수 내부에서 수행하도록 하는 방법도 고려할 필요 있음

Reply:
말씀하신대로 `memberFindDTO.getType()` 을 한 번만 1회만 수행 하는 코드가 더 좋은 코드라고 생각합니다.
MemberController의 find는 제안해주신 것처럼 변경해서 반영하도록 하겠습니다.

memberService 또는 MemberRepository의 find 코드를 하나로 통합하는 방법에 대해서는 조금 회의적입니다.
사용자가 회원 가입을 하는 동작을 예로 들었을 때, 아이디 중복 검사를 하는 과정이 필요합니다.
이 때는 정확히 findByUserId 함수의 동작 만을 필요로 하기 때문에 이 둘의 동작을 분리 시키고 Controller단에서 필요에 따라 합치는 것이 코드의 가독성 측면에서 더 좋지 않을까 생각합니다.

이에 대한 생각은 어떤지 의견 주시면 감사하겠습니다.